h2o-core/src/main/java/water/fvec/Vec.java

```diff
@@ -1539,7 +1539,7 @@ static private Key init_key() {
     public int reserveKeys(final int n) {
       AddVecs2GroupTsk tsk = new AddVecs2GroupTsk(_key, n);
       tsk.invoke(_key);
-      return tsk._n;
```

Comment:
@michalkurka changes make sense in terms of clarity, but still not 100% clear to me why it was racing here:  there's one `tsk` per thread, the only object that can be modified concurrently here is the `VecGroup`, correct?

Reply:
The issue here is that the atomic operation should not mutate any state that is not purely used for output. In this case, `_n` is mutated (old._len) is stored, however, when the automatic operation fails and is re-run we lost the original `_n` and the task no longer know how many keys to allocate! Very ugly.

Reply:
> when the atomic operation fails (due to racy update) and is re-run...

Oh, I see, makes sense then, thanks!